### ISSUE TITLE
ftplugin needs to be manually triggered

### ISSUE THREAD

**Issue by bdcaf** [neovim/neovim#19107]
### Neovim version (nvim -v)

v0.7.0

### Vim (not Nvim) behaves the same?

no, 8.2.5150

### Operating system/version

macOS 12.4

### Terminal name/version

iTerm2 3.4.15

### $TERM environment variable

xterm-256color

### Installation

homebrew

### How to reproduce the issue

_Create a test ftplugin:_
in an empty folder
```
mkdir ftplugin
cat<<END > ftplugin/c.vim
echom "sourcing " . expand('<sfile>')
END 
```

The structure is then:
```
> tree
.
└── ftplugin
    └── c.vim
```

open a c file:
`nvim --clean --cmd "filetype plugin on" --cmd "set rtp+=." test.c`

### Expected behavior

There should be a message `sourcing /Users/.../ftplugin/c.vim`

Vim behaves like this when called with
`vim --clean --cmd "filetype plugin on" --cmd "set rtp+=." test.c`

### Actual behavior

There is no message that the ftplugin was loaded. However `:set ft?` shows it  correctly  as `filetype=c`.  However if I manually enter `:filetype detect` - the expected message is printed.

Excuses if this is something trivial - but I couldn't find anything in the documentation that some extra configuration would be needed for ftplugins.

**Comment by clason** [comment]
https://github.com/neovim/neovim/wiki/FAQ#calling-inputlist-echomsg--in-filetype-plugins-and-autocmd-does-not-work

**Comment by clason** [comment]
Duplicate of #16140

**Comment by bdcaf** [comment]
Thanks that solves my issue. Wow, I expected it would be something trivial, but it didn't occur to me to research for `echom`.

**Comment by justinmk** [comment]
I thought we fixed this or at least improved some cases in 0.8 and possibly 0.7.2 ?

**Comment by clason** [comment]
I think only for autocommands: https://github.com/neovim/neovim/pull/18251?

**Comment by clason** [comment]
No, the fix also applies to ftplugin `echom`. And, yes, it has been backported, so the issue should be fixed in 0.7.2!
